plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -473,8 +474,9 @@ private static Map<String, Integer> getArrayColumnDimensions(Connection connecti
                 "AND tbl.relname = ? " +
                 "AND attyp.typcategory = 'A' ";
         try (PreparedStatement statement = connection.prepareStatement(sql)) {
-            statement.setString(1, tableHandle.getSchemaName());
-            statement.setString(2, tableHandle.getTableName());
+            RemoteTableName remoteTableName = tableHandle.asPlainTable().getRemoteTableName();
```

Comment:
The callers of this method ensure the handle is always a named relation so we can use `getRequiredNamedRelation`. Same in `getColumns` above.

(No changes requested for now - I'm confused myself about when to use one vs the other).

Reply:
Good catch!